Remove Instrumenter.Default (#1610)
open-telemetry/opentelemetry-java-instrumentation: docs/contributing/muzzle.md

is stored in the instrumentation module class in the method `InstrumentationModule#getMuzzleReferenceMatcher()`
and is shared between all type instrumentations. The bytecode of this method (basically an array of
`Reference` builder calls) is generated automatically by the ByteBuddy plugin using an ASM code visitor.

The source code of the compile-time plugin is located in the `javaagent-tooling` module,
package `io.opentelemetry.javaagent.tooling.muzzle.collector`.

### Runtime reference matching

The runtime reference matching process is implemented as a ByteBuddy matcher in `InstrumentationModule`.
`MuzzleMatcher` uses the `getMuzzleReferenceMatcher()` method generated during the compilation phase
to verify that the class loader of the instrumented type has all necessary symbols (classes,
methods, fields). If the `ReferenceMatcher` finds any mismatch between collected references and the
actual application classpath types the whole instrumentation is discarded.

It is worth noting that because the muzzle check is expensive, it is only performed after a match
has been made by the `InstrumentationModule#classLoaderMatcher()` and `TypeInstrumentation#typeMatcher()`

> matchers. The result of muzzle matcher is cached per each classloader, so that it is only executed
> once fo the whole instrumentation module.

Reviewer: ```suggestion
matchers. The result of muzzle matcher is cached per classloader, so that it is only executed
once for the whole instrumentation module.
```
